chia/wallet/wallet_node_api.py

```diff
@@ -53,8 +53,12 @@ async def new_peak_wallet(self, peak: wallet_protocol.NewPeakWallet, peer: WSChi
                 peer for peer in full_node_connections if not self.wallet_node.is_trusted(peer) and not peer.closed
             ]
 
-            # Check for untrusted peers first to avoid calling is_peer_synced if not required
-            if len(untrusted_peers) > 0 and await self.wallet_node.is_peer_synced(peer, peak.height):
+            # Check for untrusted peers to avoid fetching the timestamp if not required
+            if len(untrusted_peers) > 0:
+                timestamp = await self.wallet_node.get_timestamp_for_height_from_peer(peak.height, peer)
```

Comment:
if we have a trusted node, do we always have access to the timestamp then?

Reply:
This logic explicitly request the timestamp from the trusted peer to check if its synced. If its synced we disconnect from untrusted peers. This `if len(untrusted_peers) > 0:` is only to avoid requesting the timestamp if there are anyway no untrusted peers to disconnect from.